.github/workflows/sqlite_orm.yml

```diff
@@ -0,0 +1,63 @@
+name: sqlite_orm
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+
+  build:
+    runs-on: self-hosted
+    if: |
+      (github.event_name == 'workflow_dispatch') ||
+      (github.event_name == 'push' && github.ref == 'refs/heads/main') ||
+      (github.event_name == 'pull_request' && contains(github.event.pull_request.title, 'sqlite_orm'))
+    steps:
+    - name: Checkout build-files
+      uses: actions/checkout@v4
+      with:
+        path: build-files
+
+    - name: Checkout sqlite_orm
+      uses: actions/checkout@v4
+      with:
+        repository: fnc12/sqlite_orm
+        path: sqlite_orm
+        ref: v1.9
+
+    - name: Download SDP 8.0
+      env:
+        LICENSE_KEY: ${{ secrets.LICENSE_KEY }}
+        MYQNX_USER: ${{ secrets.MYQNX_USER }}
+        MYQNX_PASSWORD: ${{ secrets.MYQNX_PASSWORD }}
+      run: |
+        echo "Downloading QNX Software Center ..."
+        mkdir ${{ github.workspace }}/.qnx
+        curl -v --cookie-jar ${{ github.workspace }}/.qnx/myqnxcookies.auth --form "userlogin=$MYQNX_USER" --form "password=$MYQNX_PASSWORD" -X POST https://www.qnx.com/account/login.html > login_response.html
+        curl -v -L --cookie ${{ github.workspace }}/.qnx/myqnxcookies.auth https://www.qnx.com/download/download/77351/qnx-setup-2.0.3-202408131717-linux.run > qnx-setup-lin.run
+        chmod a+x qnx-setup-lin.run
+        ./qnx-setup-lin.run force-override disable-auto-start agree-to-license-terms ${{ github.workspace }}/qnxinstall
+        echo "Installing License ..."
+        ${{ github.workspace }}/qnxinstall/qnxsoftwarecenter/qnxsoftwarecenter_clt -syncLicenseKeys -myqnx.user="$MYQNX_USER" -myqnx.password="$MYQNX_PASSWORD" -addLicenseKey $LICENSE_KEY
+        cp -r ~/.qnx/license ${{ github.workspace }}/.qnx
+        echo "Downloading QNX SDP ..."
+        ${{ github.workspace }}/qnxinstall/qnxsoftwarecenter/qnxsoftwarecenter_clt -mirror -cleanInstall -destination ${{ github.workspace }}/qnx800 -installBaseline com.qnx.qnx800 -myqnx.user="$MYQNX_USER" -myqnx.password="$MYQNX_PASSWORD"
+
+    - name: Build the Docker image
+      run: |
+        cd build-files/docker && ./docker-build-qnx-image.sh
+
+    - name: Build sqlite_orm
+      uses: addnab/docker-run-action@v3
+      with:
+        image: qnx800:latest
+        options: -v ${{ github.workspace }}:/home/runner
+        shell: bash
+        run: |
+          source ~/qnx800/qnxsdp-env.sh
+          QNX_PROJECT_ROOT="$(pwd)/sqlite_orm" make -C build-files/ports/sqlite_orm JLEVEL=4 install
```

Comment:
Should we use JLEVEL=${nproc} for test workflows since we are going to get dedicated machines soon?